blocks/api/registration.js

```diff
@@ -12,29 +12,34 @@ import { applyFilters } from '@wordpress/hooks';
 import { select, dispatch } from '@wordpress/data';
 import deprecated from '@wordpress/deprecated';
 
+/**
+ * Internal dependencies
+ */
+import { normalizeIconObject } from './utils';
+
 /**
  * Defined behavior of a block type.
  *
  * @typedef {WPBlockType}
  *
- * @property {string}             name       Block's namespaced name.
- * @property {string}             title      Human-readable label for a block.
- *                                           Shown in the block inserter.
- * @property {string}             category   Category classification of block,
- *                                           impacting where block is shown in
- *                                           inserter results.
- * @property {(string|WPElement)} icon       Slug of the Dashicon to be shown
- *                                           as the icon for the block in the
- *                                           inserter, or element.
- * @property {?string[]}          keywords   Additional keywords to produce
- *                                           block as inserter search result.
- * @property {?Object}            attributes Block attributes.
- * @property {Function}           save       Serialize behavior of a block,
- *                                           returning an element describing
- *                                           structure of the block's post
- *                                           content markup.
- * @property {WPComponent}        edit       Component rendering element to be
- *                                           interacted with in an editor.
+ * @property {string}                    name       Block's namespaced name.
+ * @property {string}                    title      Human-readable label for a block.
+ *                                                  Shown in the block inserter.
+ * @property {string}                    category   Category classification of block,
+ *                                                  impacting where block is shown in
+ *                                                  inserter results.
+ * @property {(Object|string|WPElement)} icon       Slug of the Dashicon to be shown
```

Comment:
Thinking at some point we might just want to `@typedef` a `WPIcon`, since it's used pretty broadly now.

http://usejsdoc.org/tags-typedef.html

Related: https://github.com/WordPress/gutenberg/pull/5965#issuecomment-383557878 (cc @ajitbohra)

Reply:
Yes, here what we have is (Object|WPIcon) I think when we have generic Icon component in @wordpress/components we should create the type there so the type can be imported from this module to other modules.

Reply:
Note that the plan is to move away from UI icons for blocks and into custom svgs, so this will become more of a fallback.